### PR TITLE
feat(serve): add --inspect flag

### PR DESCRIPTION
**Pull request by aleclarson**
### Description

Add an `--inspect` flag to `firebase serve`.

When defined, the `debugPort` of `FunctionsEmulator` becomes 9229 or the specified port (eg: `--inspect=9230`).

### Scenarios Tested

N/A

### Sample Commands

```sh
firebase serve --inspect
firebase serve --inspect=9230
```

### CODE REVIEW

**Comment by samtstern** [comment]
@bkendall thoughts on this?  I know we're not supposed to be improving or investing in `serve` anymore but this seems harmless, and there are a few Hosting edge cases where `serve` is still superior to `emulators:start`

@aleclarson thanks for this PR, a few comments:

1. Please make this flag `--inspect-functions` to match what we have with `emulators:start`
2. Is there a reason you still prefer to use `serve` and not the `emulators:` commands?

**Comment by bkendall** [comment]
My preference would be to keep this in `emulators:start` and not add it to `serve`. I too would like to know about (2) in Sam's comment. 😄

**Comment by aleclarson** [comment]
2. Is there documentation on the differences between them? I'm using `serve` because I don't understand why switching is necessary.

**Comment by samtstern** [comment]
@aleclarson the difference is that `serve` starts up individual emulators which are mostly not connected to each other. It was designed for Hosting, then we added on Functions, and then we added on the two database emulators.  We consider `serve` to be pretty much deprecated, we just don't want to break it because so many Hosting developers rely on it.

`emulators:start` does everything `serve` does and more, such as:

- Connects all the emulators together so you can, for example, trigger local Cloud Functions from the Realtime Database emulator
- Centralizes emulator configuration in `firebase.json` through `firebase init emulators`
- Starts up a local UI which allows you to visually interact with the emulators
- Supports importing/exporting data and other complex flows
- .... more!

So basically in  99/100 cases if you just replace `firebase serve` with `firebase emulators:start` everything will keep working and you'll get some great new features as well.

**Comment by aleclarson** [comment]
Why's there no `--port` flag for `emulators:start`?

**Comment by samtstern** [comment]
@aleclarson because there are many emulators a single `--port` flag no longer makes sense.  That's all handled through `firebase.json` now.  So if you want to change the Hosting emulator port:

**firebase.json**
```js
{
  "emulators": {
    "hosting": {
      "port": 3003 // Example
    }
}
```
